docs/fp-SDK-guide.md

```diff
@@ -0,0 +1,414 @@
+# Finality Provider SDK Developer Guide
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Core constructor](#2-core-constructor)
+   1. [Default Usage](#21-default-usage)
+   2. [Custom Usage](#22-custom-usage)
+3. [Interfaces](#3-interfaces)
+   1. [Communication Level Interfaces](#31-communication-level-interfaces)
+      1. [ConsumerController Interface](#311-consumercontroller-interface)
+   2. [Service Level Interfaces](#32-service-level-interfaces)
+      1. [Block Polling Interface](#321-blockpolling-interface)
+      2. [Randomness Committer Interface](#322-randomnesscommitter-interface)
+      3. [Finality Signature Submitter Interface](#323-finalitysignaturesubmitter-interface)
+      4. [Height Determiner Interface](#324-heightdeterminer-interface)
+4. [Instantiation](#4-instantiation)
+   1. [Using Built-in Implementations](#41-using-built-in-implementations)
+   2. [Using Custom Implementations](#42-using-custom-implementations)
+5. [Startup](#5-startup)
+
+## 1. Introduction
+
+The Finality Provider SDK enables Bitcoin Supercharged Networks (BSNs) to 
+integrate with Babylon's Bitcoin-secured finality infrastructure as a flexible 
+library. It provides finality guarantees for consumer chain blocks through 
```

Comment:
Do you think it's useful to mention here what are key things sdk do:
1. Block querying
2. Randomness generation
3. Finality signature generation
4. Submitting data